iothub/service/tests/SerializationTests.cs

```diff
@@ -12,69 +15,249 @@ namespace Microsoft.Azure.Devices.Tests
         }
 
         [TestMethod]
-        public void Configuration_TestNullSchema_Ok()
+        public void FeedbackRecord_JsonParse_Ok()
         {
-            const string expectedDeviceId = "aa";
-            const string expectedSchemaVersion = "1.0";
-            const string jsonString = @"
-{
-  ""id"": ""aa"",
-  ""content"": {
-    ""modulesContent"": {
-        ""$edgeAgent"": {
-            ""properties.desired"": {
-            }
+            // arrange - act
+            var feedbackRecord = new FeedbackRecord
+            {
+                OriginalMessageId = "1",
+                DeviceGenerationId = "2",
+                DeviceId = "testDeviceId",
+                EnqueuedOnUtc = new DateTimeOffset(2023, 1, 20, 8, 6, 32,
+                                                new TimeSpan(1, 0, 0)),
+                StatusCode = FeedbackStatusCode.Success,
+                Description = "Success"
+            };
+            string feedbackRecordSerialized = JsonConvert.SerializeObject(feedbackRecord);
+
+            // assert
+            FeedbackRecord fr = JsonConvert.DeserializeObject<FeedbackRecord>(feedbackRecordSerialized);
+            fr.Should().BeEquivalentTo(feedbackRecord);
         }
-    }
-  }
-}";
-            Configuration configuration = JsonConvert.DeserializeObject<Configuration>(jsonString);
-            configuration.Id.Should().Be(expectedDeviceId);
-            configuration.SchemaVersion.Should().Be(expectedSchemaVersion);
+
+        [TestMethod]
+        public void FileUploadNotification_JsonParse_Ok()
+        {
+            // arrange - act
+            var fileUploadNotification = new FileUploadNotification
+            {
+                DeviceId = "testDeviceId",
+                BlobName = "testBlob",
+                BlobUriPath = new Uri("https://myaccount.blob.core.windows.net"),
+                BlobSizeInBytes = 50,
+                LastUpdatedOnUtc = new DateTimeOffset(2023, 1, 19, 8, 7, 32,
+                                                new TimeSpan(1, 0, 0)),
+                EnqueuedOnUtc = new DateTimeOffset(2023, 1, 20, 8, 6, 32,
+                                                new TimeSpan(1, 0, 0))
+            };
+
+            string fileUploadNotificationSerialized = JsonConvert.SerializeObject(fileUploadNotification);
+
+            FileUploadNotification fun = JsonConvert.DeserializeObject<FileUploadNotification>(fileUploadNotificationSerialized);
+
+            // assert
+            fun.Should().BeEquivalentTo(fileUploadNotification);
+        }
+
+        [TestMethod]
+        public void BasicDigitalTwin_JsonParse_Ok()
+        {
+            // arrange - act
+            var digitalTwinMetadata = new DigitalTwinMetadata
+            {
+                ModelId = "modelId1234"
+            };
+
+            var basicDigitalTwin = new BasicDigitalTwin
+            {
+                Id = "twinId1234",
+                Metadata = digitalTwinMetadata
+            };
```

Comment:
Might as well in-line this init as it produces fewer objects for a reader to track.

```suggestion
            var basicDigitalTwin = new BasicDigitalTwin
            {
                Id = "twinId1234",
                Metadata = new DigitalTwinMetadata
                {
                    ModelId = "modelId1234"
                },
            };
```